### DISCUSSION TITLE
Creating parent query combining individual entities

### DISCUSSION THREAD

**Discussion by Palak-K9** [neo4j-graphql/neo4j-graphql-java#313] [Q&A] [answered]
I am trying to build and application using the example - [Spring Boot Graphql Example](https://github.com/neo4j-graphql/neo4j-graphql-java/tree/master/examples/graphql-spring-boot)

I have **Neo4j Database** in which there are various node types like **Fruit, Sweet, Vegetable**. I have defined thier schema in **neo4j.graphql file**. and while running graphiql on browser, I can directly query these nodes with conditions.
For example :-

```{  
  fruits{
    name
    quantity
  }
 }
```
 
 This query gives me the fruit nodes with the mentioned fields.
 I can query the individual entities but I want to get these entities under one name/query. I have tried defining the following in no4j.graphql
 
```
 type Query {
      getItems: [Item]
   }
   
  type Item{
     fruits: [Fruit]
     vegetables: [Vegetable]
     sweets: [Sweet]
     }
 ```
 
 But this is not working. Could someone help me how can I acheive this

**Comment by Andy2003** [comment]
Can you provide your schema please!

**Reply by Palak-K9** [reply]
````
type Query {
      getItems: [Item]
   }
   
type Item{
     fruits: [Fruit]
     vegetables: [Vegetable]
     sweets: [Sweet]
   }
     
type Fruit{
  id: Int
  name: String!
  quantity: Int!
 }
 
 type Vegetable{
  id: Int
  name: String!
  quantity: Int!
 }
 
 type Sweet{
  id: Int
  name: String!
  quantity: Int!
 }

**Reply by Andy2003** [reply]
what about using an interface?

```graphql
interface Item{
     id: Int
  name: String!
  quantity: Int!
}

type Fruit implements Item {
  id: Int
  name: String!
  quantity: Int!
 }

 type Vegetable implements Item {
  id: Int
  name: String!
  quantity: Int!
 }

 type Sweet implements Item {
  id: Int
  name: String!
  quantity: Int!
 }
```

**Reply by Palak-K9** [reply]
But I want result set like the following -
```
getItems{
  fruits{
     [ { id:1
        name......}, {..}, ....] }
vegetables{
     [ { id:1
        name......}, {..}, ....]}
sweets{
     [ { id:1
        name......}, {..}, ....]}
}
```
And all items might not have same properties.

**Reply by Andy2003** [reply]
I think you are asking for Unions, which are currently not yet implemented. It will look like this https://neo4j.com/docs/graphql/current/type-definitions/types/unions/

You can also try to run multiple queries in one request:

```graphql
{
    fruit {
      name
    }
    vegetable {
      name
    }
    sweet {
      name
    }
}
```

**Reply by Palak-K9** [reply]
Yes, I am currently running different queries in one request, but I want to keep them under one query.
Like if I call getItems
And try to get fruits under it. i dont want to run fruits, vegetables and sweets as individual queries but run them under getItems. is this possible?

**Reply by Andy2003** [reply, marked as answer]
Adding an extra layer (wrapper) around a field is nut supported by this library.
You can try to write a custom `@cypher` to meet your needs.

**Reply by Palak-K9** [reply]
Okay, will try
